### PR TITLE
tests for factories to containers migration

### PR DESCRIPTION
**Pull request by basz**

### CODE REVIEW

**Comment by bakura10** [comment]
Thanks a lot for the work!

Let me know once this is mergeable :).

**Comment by basz** [comment]
there! please review, but I think it is another baby step... the tests (factory/container) run, but I think some stuff has to be removed there (mark one as skipped and i placed a few comments). I don't guarantee they do what they should do :-/

**Comment by basz** [comment]
done for the day, lets review and merge or this will get too big to handle.

File: docs/03. Role providers.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ If you are more confident with flat RBAC, the previous config can be re-written
 return [
     'zfc_rbac' => [
         'role_provider' => [
-            'ZfcRbac\Role\InMemoryRoleProvider' => [
+            \ZfcRbac\Role\InMemoryRoleProvider::class => [
```
Comment (bakura10) [inline docs/03. Role providers.md:96]: Thanks for updating the doc, however let's keep that for the end, as I suspect we'll need to do some major architectural changes :).

File: test/Container/AssertionPluginManagerFactoryTest.php [hunk 2]
```diff
@@ -19,26 +19,33 @@
...
 
         $factory       = new AssertionPluginManagerFactory();
-        $pluginManager = $factory->createService($serviceManager);
+        $pluginManager = $factory($serviceManager, 'requestedName');
 
-        $this->assertInstanceOf('ZfcRbac\Assertion\AssertionPluginManager', $pluginManager);
-        $this->assertSame($serviceManager, $pluginManager->getServiceLocator());
+        $this->assertInstanceOf(AssertionPluginManager::class, $pluginManager);
+
+        # @todo remove after review
```
Comment (bakura10) [inline test/Container/AssertionPluginManagerFactoryTest.php:44]: You're right :).

File: test/Container/RoleServiceFactoryTest.php [hunk 3]
```diff
@@ -18,69 +18,81 @@
...
         $serviceManager->setService(
-            'ZfcRbac\Identity\AuthenticationProvider',
-            $this->getMock('ZfcRbac\Identity\IdentityProviderInterface')
+            AuthenticationProvider::class,
+            $this->getMock(IdentityProviderInterface::class)
         );
 
-        $traversalStrategy = $this->getMock('Rbac\Traversal\Strategy\TraversalStrategyInterface');
-        $rbac              = $this->getMock('Rbac\Rbac', [], [], '', false);
+        $traversalStrategy = $this->getMock(TraversalStrategyInterface::class);
```
Comment (bakura10) [inline test/Container/RoleServiceFactoryTest.php:63]: You're right, TraversalStraetgyInterface has been removed from Rbac, that's something that will need to be updated in RoleService, but let's do that after :).

File: test/Util/ServiceManagerFactory.php [hunk 4]
```diff
@@ -1,76 +0,0 @@
...
-use Zend\Mvc\Service\ServiceManagerConfig;
-use Zend\ServiceManager\ServiceManager;
-
-/**
- * Base test case to be used when a new service manager instance is required
- *
- * @license MIT
- * @author  Marco Pivetta <ocramius@gmail.com>
- */
-abstract class ServiceManagerFactory
```
Comment (bakura10) [inline test/Util/ServiceManagerFactory.php:30]: Yeah \o/.

**Comment by bakura10** [comment]
Thanks. I didn't read everything closely, but I'm going to merge so we can start actually think a bit more about the architecture.

**Comment by basz** [comment]
if you let me I will continue on attending to the tests - it's a good chance for me to get intimate with rbac and zf

**Pull request merged**